### ISSUE TITLE
[Core][Metrics] Prometheus-client not working with the latest version. 

### ISSUE THREAD

**Issue by rkooo567** [ray-project/ray#25523]
### What happened + What you expected to happen

When the latest Prometheus-client is used, the following error seems to occur based on the report. 

```
  File "/home/ray/anaconda3/lib/python3.7/socket.py", line 752, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known
```

Although we restrict the Prometheus client version in the master, it is ideal to fix this since it seems to be pretty common to download wrong dependencies (it's quite common in python). Or we should at least raise a better error message.

### Versions / Dependencies

Master

### Reproduction script

Run the lastest prometheus client

### Issue Severity

Low: It annoys or frustrates me.

**Comment by rkooo567** [comment]
@alanwguo said it is fixed. @rkooo567 to verify and close.

**Comment by galenhwang** [comment]
I believe the fix was https://github.com/ray-project/ray/pull/23766?

**Comment by rkooo567** [comment]
I don't think that fixes the root cause. It just mitigates the issue. It's ideal we don't restrict the version requirement.

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
